upload.php

```diff
@@ -129,33 +128,22 @@ function emr_delete_current_files( $current_file ) {
 		$new_guid      = str_replace( $current_filename, $new_filename, $current_guid );
 
 		// Update database file name
-		$sql = $wpdb->prepare(
-			"UPDATE $table_name SET post_title = '$new_filetitle', post_name = '$new_filetitle', guid = '$new_guid', post_mime_type = '$new_filetype' WHERE ID = %d;",
-			(int) $_POST["ID"]
-		);
-		$wpdb->query( $sql );
-
-		// Update the postmeta file name
-
-		// Get old postmeta _wp_attached_file
-		$sql = $wpdb->prepare(
-			"SELECT meta_value FROM $postmeta_table_name WHERE meta_key = '_wp_attached_file' AND post_id = %d;",
-			(int) $_POST["ID"]
+		$post_data = array(
+			'ID' => absint( $_POST['ID'] ),
+			'post_title' => $new_filetitle,
+			'post_name' => $new_filetitle,
+			'guid' => $new_guid,
+			'post_mime_type' => $new_filetype
 		);
 
-		$old_meta_name = $wpdb->get_row( $sql, ARRAY_A );
-		$old_meta_name = $old_meta_name["meta_value"];
+		$ret = wp_update_post( $post_data );
```

Comment:
Unfortunately using `wp_update_post()` fails to update the guid as it's protected. The neatest not so neat way I found to do it was like this:

```php
// WordPress overrides if the guid is set which it will because the media is downloaded last
$override_guid = function ( $data, $postarr ) use ( $url ) {
    $data['guid'] = $url;
    return $data;
};

add_filter( 'wp_insert_attachment_data', $override_guid, 10, 2 );

wp_update_post( array(
    'ID'             => $post_id,
    'post_mime_type' => $type,
    'guid'           => $url,
    'post_title'     => wp_filter_kses( $title ),
) );

remove_filter( 'wp_insert_attachment_data', $override_guid, 10 );
```